gitrevise/merge.py

```diff
@@ -28,6 +28,12 @@
 DEFAULT_CONFLICT_MARKER_SIZE = 7
 
 
+def get_conflict_marker_size(__repo: Repository, __file: Path) -> int:
+    # TODO: Determine on a per-file basis by its `conflict-marker-size` attribute.
+    # See ll_merge_marker_size in git/ll-merge.c
```

Comment:
Regarding your comment about performance concerns: here's a possible way to avoid them:
1. In merge_files(), run "git merge-file", without passing a marker size
2. Only if the above "git merge-file" failed, ask "git check-attr" for the conflict marker size, and re-run "git merge-file"

In the common case (no conflict) we're as fast as ever.

I often use git-revise for rewrites that cause the same conflicts - that's
where rerere kicks in. However, there are typically very few of those
conflicts, so I don't think that matters. Also that's not the typical use case.

Reply:
Doing an optimistic merge is a good idea to consider. I'd worry a little about being tempted (or tempting others) to look for or save cached resolutions for both marker sizes, which could lead to inconsistency. Probably avoidable, though! Or not a big deal.

I do wonder what the actual performance impact of batch-querying `git check-attrs` for each un-encountered entry per tree, e.g. using a persistent process like `git cat-file` is using. I suspect that the performance difference between that and re-merging is pretty small. (I was mainly leery of my hamfisted query-each-entry-individually 😄).

Reply:
> Probably avoidable, though!

Yeah, merge_files() would do two runs of "git merge-file" and if the first failed, callers would only ever see the results of the second.

Using a persistent process for `git check-attrs` sounds great, and there are probably other reasons to use that: we can detect attributes like file encoding and whether a file is binary.